messaging/src/test/java/org/axonframework/eventhandling/AnnotationEventHandlerAdapterTest.java

```diff
@@ -125,6 +129,31 @@ void canHandleTypeDoesNotReturnResetHandlers() {
         assertFalse(testSubject.canHandleType(Integer.class));
     }
 
+    @Test
+    void replayNotSupportedOnSingleHandler() {
+        SingleReplayBlockingHandler handler = new SingleReplayBlockingHandler();
+        testSubject = new AnnotationEventHandlerAdapter(handler, parameterResolverFactory);
+
+        assertTrue(testSubject.supportsReset());
+    }
+
+    @Test
+    void replayNotSupportedOnClassLevel() {
+        ReplayBlockedOnClassLevelHandler handler = new ReplayBlockedOnClassLevelHandler();
+        testSubject = new AnnotationEventHandlerAdapter(handler, parameterResolverFactory);
+
+        assertFalse(testSubject.supportsReset());
+    }
+
+    @Test
+    void replayNotSupportedOnClassLevelWithHandlerLevelOverride() {
+        ReplayBlockedOnClassLevelWithReplayCapableHandler handler = new ReplayBlockedOnClassLevelWithReplayCapableHandler();
+        testSubject = new AnnotationEventHandlerAdapter(handler, parameterResolverFactory);
+
+        assertTrue(testSubject.supportsReset());
+    }
+
```

Comment:
Nit: we could perhaps add a test case using an `@AllowReplay(value = false)` annotation.